playground/assets/vite.config.js

```diff
@@ -1,21 +1,22 @@
-const path = require('node:path')
+// @ts-check
+import path from 'node:path'
+import { defineConfig } from 'vite'
 
-/**
- * @type {import('vite').UserConfig}
- */
-module.exports = {
-  base: '/foo',
-  publicDir: 'static',
-  resolve: {
-    alias: {
-      '@': path.resolve(__dirname, 'nested'),
+export default /** @type {import('vite').UserConfig} */ (
+  defineConfig({
```

Comment:
I don't think we need this jsdoc as we're using `defineConfig` here.

Reply:
Indeed it does need and that's why https://github.com/vitejs/vite/pull/12021 was filed. It's reverted now so we had to keep this 😂. Only re-used config files are added explicit type annotation for now.

---

error without type annotation

```
Property 'build' does not exist on type 'UserConfigExport'.
  Property 'build' does not exist on type 'Promise<UserConfig>'.ts(2339)
```

Reply:
Interesting, when I tested locally I didn't get this error, maybe I can try again. Just to confirm, are you using the workspace TypeScript version, instead of VSCode's bundled TypeScript version?

Reply:
I tested it with `pnpm tsc --noEmit` so it does will fail in CI. 😛

> when I tested locally I didn't get this error,

I guess you're running with a `pnpm dev` of vite. In this mode, types are in `vite/dist/dist/node`(double dist 🤨) instead of `"types": "./dist/node/index.d.ts"`. I'm not sure whether it's a workaround or a bug 🤔. But I think it should be fixed anyway.

Reply:
Maybe not relevant to this PR. But fix the missing d.ts in dev mode in another PR https://github.com/vitejs/vite/compare/main...fi3ework:build-corrent-dev-dts?expand=1

Reply:
Ah I see it now. The error is happening in other files:

<details>
<summary>error log</summary>

```
assets/vite.config-relative-base.js:8:19 - error TS2339: Property 'build' does not exist on type 'UserConfigExport'.
  Property 'build' does not exist on type 'Promise<UserConfig>'.

8     ...baseConfig.build,
                    ~~~~~

assets/vite.config-runtime-base.js:13:19 - error TS2339: Property 'build' does not exist on type 'UserConfigExport'.
  Property 'build' does not exist on type 'Promise<UserConfig>'.

13     ...baseConfig.build,
                     ~~~~~


Found 2 errors in 2 files.
```
</details>

Perhaps we can typecast in the other files instead, or typecast in this file like this?

```js
/** @type {import('vite').UserConfig} */
// @ts-expect-error typecast
export default defineConfig({
```

I think the main goal is to avoid the indent here as that changes git history for all the lines that makes debugging slightly harder in the future.

Reply:
updated.
I would prefer to add annotations in the source file as the importers might be more than the importees. We could make the code lesser in this way.

Reply:
yeah. i think using `ts-expect-error` is good too since when we figure out adding overloads in a non-breaking way (or if we do overloads in next major), it should help clean up these comments.